routing/routing_tests/turns_tts_text_tests.cpp

```diff
@@ -156,6 +156,77 @@ UNIT_TEST(GetTtsTextTest)
+      \"then\":\"Then.\",\
+      \"make_a_right_turn\":\"Make a right turn.\",\
+      \"make_a_left_turn\":\"Make a left turn.\",\
+      \"dist_direction_onto_street\":\"%1$s %2$s onto %3$s\",\
+      \"you_have_reached_the_destination\":\"You have reached the destination.\"\
+      }";
+
+  string const jaShortJson =
+      "\
+      {\
+      \"in_300_meters\":\"三百メートル先\",\
+      \"in_500_meters\":\"五百メートル先\",\
+      \"then\":\"その先\",\
+      \"make_a_right_turn\":\"右折です。\",\
+      \"make_a_left_turn\":\"左折です。\",\
+      \"make_a_right_turn_street\":\"右折し\",\
+      \"make_a_left_turn_street\":\"左折し\",\
+      \"dist_direction_onto_street\":\"%1$s%2$s %3$s に入ります\",\
+      \"you_have_reached_the_destination\":\"到着。\"\
+      }";
+
+  string const faShortJson =
+      "\
+      {\
+      \"in_300_meters\":\"ﺩﺭ ﺲﯿﺻﺩ ﻢﺗﺮﯾ\",\
+      \"in_500_meters\":\"ﺩﺭ ﭖﺎﻨﺻﺩ ﻢﺗﺮﯾ\",\
+      \"then\":\"ﺲﭙﺳ\",\
+      \"make_a_right_turn\":\"ﺐﻫ ﺭﺎﺴﺗ ﺐﭙﯿﭽﯾﺩ.\",\
+      \"make_a_left_turn\":\"ﺐﻫ ﭻﭘ ﺐﭙﯿﭽﯾﺩ.\",\
+      \"dist_direction_onto_street\":\"%1$s %2$s ﺐﻫ %3$s\",\
+      \"you_have_reached_the_destination\":\"ﺶﻣﺍ ﺮﺴﯾﺪﻫ ﺎﯾﺩ.\"\
+      }";
+
+  GetTtsText getTtsText;
+  // Notification(uint32_t distanceUnits, uint8_t exitNum, bool useThenInsteadOfDistance,
+  //    CarDirection turnDir, Settings::Units lengthUnits, std::string nextStreet)
+  Notification const notification1(500, 0, false, CarDirection::TurnRight,
+                                   measurement_utils::Units::Metric, "Main Street");
+  Notification const notification2(300, 0, false, CarDirection::TurnLeft,
+                                   measurement_utils::Units::Metric, "Main Street");
+  Notification const notification3(300, 0, false, CarDirection::TurnLeft,
+                                   measurement_utils::Units::Metric);
+  Notification const notification4(0, 0, true, CarDirection::TurnLeft,
+                                   measurement_utils::Units::Metric);
+
+  getTtsText.ForTestingSetLocaleWithJson(engShortJson, "en");
+  TEST_EQUAL(getTtsText.GetTurnNotification(notification1), "In 500 meters Make a right turn onto Main Street", ());
+  TEST_EQUAL(getTtsText.GetTurnNotification(notification2), "In 300 meters Make a left turn onto Main Street", ());
+  TEST_EQUAL(getTtsText.GetTurnNotification(notification3), "In 300 meters. Make a left turn.", ());
+  TEST_EQUAL(getTtsText.GetTurnNotification(notification4), "Then. Make a left turn.", ());
+
+  getTtsText.ForTestingSetLocaleWithJson(jaShortJson, "ja");
+  TEST_EQUAL(getTtsText.GetTurnNotification(notification1), "五百メートル先右折し Main Street に入ります", ());
+  TEST_EQUAL(getTtsText.GetTurnNotification(notification2), "三百メートル先左折し Main Street に入ります", ());
+  TEST_EQUAL(getTtsText.GetTurnNotification(notification3), "三百メートル先 左折です。", ()); // note the extraneous space here due to + " " +
+  TEST_EQUAL(getTtsText.GetTurnNotification(notification4), "その先 左折です。", ()); // note the extraneous space here due to + " " +
```

Comment:
There should be no spaces in those strings, right?

Reply:
It's a hard situation, there have always been spaces in these strings because the algorithm has always been `dist_str + " " + dir_str` and I'm fixing that with the new _street formatter but to fix it for all other TTS output we would have to refactor and use formatter strings for all those as well. As far as I can tell it has no effect on the spoken output so it's just a testing quirk for languages that don't require spaces between words.

Reply:
A proper implementation would take into account the language and if it has or does not have spaces to delimit words.
It is also another argument to use as few placeholders as possible in favor of some partial duplication.

Reply:
Given that it hasn't caused any problems yet and I'm not changing that part of the code, maybe we can implement these changes as-is for now and move towards a more unified placeholder-based format string methodology as we go on. That's the ultimately best answer, better than plain string concatenation.